### DISCUSSION TITLE
Workspace parameters for Citrix.Workspace.LTSR not working

### DISCUSSION THREAD

**Discussion by dlundahl91** [Romanitho/Winget-Install#116] [General]
Hey Romanitho!

Thanks again for this wonderful product, you are superstars.
We are trying to replace our win32 apps in Intune with winget installation of our Citrix, however have some issues getting it to work.

**This is the parameter we would install it with without using winget:**
"%~dp0CitrixWorkspaceFullInstaller.exe" /silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0="Work Account Store;https://citrix.ctxexample.net/Citrix/CTXExampleCom/discovery;On;CTX Example Com 7.xx"
It works well.

**I am writing it like this in the winget install command:**
powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0="Work Account Store;https://citrix.ctxexample.net/Citrix/CTXExampleCom/discovery;On;CTX Example Com 7.xx"\""

And it seems there is a problem with the number of "" symbols, but we need the "" symbols in the parameters, as we have spaces in our citrix store.

Can you help me out please?

Thanks a lot in advance!

**Comment by KnifMelti** [comment]
Escaping with `\` as in https://github.com/Romanitho/Winget-Install?tab=readme-ov-file#use-winget-native-parameters works well, but your extra "" because you have spaces in your store complicates it...
...change your store name maybe?

**Reply by KnifMelti** [reply]
Got the installation ok with `-AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"Work Account Store;https://citrix.ctxexample.net/Citrix/CTXExampleCom/discovery;On;CTX Example Com 7.xx\"\"\""` - if it's correct installed:

![image](https://github.com/user-attachments/assets/a8d5a37c-75a8-4826-a1aa-f7d300c35cef)

**Reply by dlundahl91** [reply]
Hi KnifMelti!

Thanks so much for taking the time and interest in helping me!

I tried your method, but changed it to our store instead of the "work account store", and other dummies I put in the text to hide the company I work for.
I am getting this error:
![image](https://github.com/user-attachments/assets/1179a2bf-d4a3-4827-867e-2c28f6de991e)


For me it seems that it doesn't properly send the command.
This is how I send it as a test:
`powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"hidden hidden hidden;https://citrix.hidden.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden Hidden Hidden 7.xx\"\"\""`

Can you tell if I am doing something wrong?

I really appreciate your help!

**Reply by KnifMelti** [reply]
> powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override "/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=""hidden hidden hidden;https://citrix.hidden.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden Hidden Hidden 7.xx""""
> 
> Can you tell if I am doing something wrong?

Yes, you're definitely doing it wrong. You must escape with `\`:
`powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"hidden hidden hidden;https://citrix.ctxexample.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden Hidden Hidden 7.xx\"\"\""`

**Reply by dlundahl91** [reply]
Sorry, I wrote it wrong in the comment! This is what I have been trying:
`powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"HIDDEN Hidden Hidden;https://citrix.hidden.net/Citrix/HIDDENHiddenHIdden/discovery;On;HIDDEN Hidden Hidden 7.xx\"\"\""`

Copying your command into my powershell gives this result:
PS C:\Users\WDAGUtilityAccount\downloads\source> `powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"hidden hidden hidden;https://citrix.ctxexample.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden Hidden Hidden 7.xx\"\"\""`


                888       888 d8b  .d8888b.           d8b
                888   o   888 Y8P d88P  Y88b          Y8P
                888  d8b  888     888    888
                888 d888b 888 888 888        888  888 888
                888d88888b888 888 888  88888 888  888 888
                88888P Y88888 888 888    888 888  888 888
                8888P   Y8888 888 Y88b  d88P Y88b 888 888
                888P     Y888 888  "Y8888P88  "Y88888 888

               https://github.com/Romanitho/Winget-Install
             https://github.com/Romanitho/Winget-Install-GUI

        _________________________________________________________


13:52:06 - ###   10/31/2024 - NEW INSTALL REQUEST   ###

13:52:07 - Running with admin rights.

13:52:07 - Checking prerequisites...
13:52:08 - Installed Winget version: v1.8.1911
13:52:08 - Checking prerequisites ended.

13:52:08 - Start Citrix.Workspace.LTSR processing...
13:52:08 - -> Citrix.Workspace.LTSR exists on Winget Repository.
13:52:09 - -> Installing Citrix.Workspace.LTSR...
13:52:09 - -> Running: "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.23.1911.0_x64__8wekyb3d8bbwe\winget.exe" install --id Citrix.Workspace.LTSR -e --accept-package-agreements --accept-source-agreements -s winget -h --override " /silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\\hidden hidden hidden

Found Citrix Workspace [Citrix.Workspace.LTSR] Version 24.2.1001.2
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...

Installer failed with exit code: 40017
13:52:19 - -> Citrix.Workspace.LTSR installation failed!
13:52:19 - Citrix.Workspace.LTSR processing finished!

13:52:20 - ###   END REQUEST   ###

https://citrix.ctxexample.net/Citrix/HiddenHiddenHidden/discovery : The term 'https://citrix.ctxexample.net/Citrix/HiddenHiddenHidden/discovery' is not recognized as the name of a
cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:208
+ ... dden hidden;https://citrix.ctxexample.net/Citrix/HiddenHiddenHidden/d ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (https://citrix....idden/discovery:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

On : The term 'On' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path
is correct and try again.
At line:1 char:274
+ ... /citrix.ctxexample.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden  ...
+                                                                ~~
    + CategoryInfo          : ObjectNotFound: (On:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Hidden : The term 'Hidden' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that
the path is correct and try again.
At line:1 char:277
+ ... .ctxexample.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden Hidden  ...
+                                                            ~~~~~~
    + CategoryInfo          : ObjectNotFound: (Hidden:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

![image](https://github.com/user-attachments/assets/2ceef344-94d3-4e3e-862f-2a4c6c7d1b5e)

**Reply by dlundahl91** [reply]
Please allow me to apologize for making three comments, and then afterwards consolidating them into one.. I got a bit confused on why it was removing my \ symbols, until I put it into `code` text.

**Reply by KnifMelti** [reply]
Well, then I can't help you anymore. The string I created works for me.

**Reply by dlundahl91** [reply]
I understand. I really appreciate your time and help with this! I apologize for being a special case.
Not sure why it won't work on my end, but it works on yours.. I am trying the exact same command as you gave me, with the same info:

`powershell.exe -File .\winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"Work Account Store;https://citrix.ctxexample.net/Citrix/CTXExampleCom/discovery;On;CTX Example Com 7.xx\"\"\""`

Doesn't work for me. Thanks again so much.

If anyone else has any ideas, I'd really appreciate it.

**Comment by KnifMelti** [comment]
Have you got the latest from https://github.com/Romanitho/Winget-Install/releases ?

`powershell.exe -File winget-install.ps1 -AppIDs "Citrix.Workspace.LTSR --override \"/silent /includeSSON /EnableCEIP=false /AutoUpdateCheck=disabled /noreboot ALLOWADDSTORE=A STORE0=\"\"hidden hidden hidden;https://citrix.ctxexample.net/Citrix/HiddenHiddenHidden/discovery;On;Hidden Hidden Hidden 7.xx\"\"\""`:

Running as SYSTEM with above string (but with `-NoProfile -ExecutionPolicy bypass -File C:\VirtualTest\winget-install.ps1`):

![image](https://github.com/user-attachments/assets/4a15790d-2043-4db1-8724-a967884b1979)
![image](https://github.com/user-attachments/assets/5c8c936e-68a6-4920-8c28-67787c69aed0)
![image](https://github.com/user-attachments/assets/46f177dd-5649-48a6-a852-81f72e3a5d9a)

Gives:

![image](https://github.com/user-attachments/assets/e64a1679-b64e-477a-8b34-1acca2a6b02a)

Can't give you anymore.

**Comment by KnifMelti** [comment]
The culprit here is that you're running this from a PowerShell CLI (regarding the quotes etc.).
This should be run from Command Prompt (Run as administrator) like SYSTEM is installing things from SCCM/Intune calling powershell.exe with parameters.
The best way is to use https://www.nirsoft.net/utils/advanced_run.html emulating a SYSTEM install

**Reply by dlundahl91** [reply]
You keep saying that you can't help me further, but you always come back as my white knight. 😍
As per your instructions I made the new intune package for testing and with your parameter I will test it today!

**Reply by dlundahl91** [reply]
Your instructions have made this installation work absolutely flawlessly!

I do now however have a new problem, which I believe has nothing at all to do with winget-install, but all to do with winget in general, and hope maybe you have some insight you could help me with...

I am installing this software with winget-install: Citrix.Workspace.LTSR

But once the program is installed, winget list sees it as Citrix.Workspace and not Citrix.Workspace.LTSR
This makes our winget-autoupdate, update it to the newest available workspace version, but we want to stay exclusively on LTSR version.

Can you help me find out some way to go about this?

I REALLY appreciate your knowledge, and I wish I could pick your brain or use you as copilot instead of actual copilot. 🫡

**Reply by KnifMelti** [reply]
Glad my contribution solved your dilemma!

First of all, a company shouldn't be using **WAU** with **Blacklist** (excluded_apps.txt).
You should get in control and use **Whitelist** (included_apps.txt), thereby making sure that all the apps have been tested for updating before they are put in the list!

Secondly, **Workspace App** nowadays has its own service for upgrading so it shouldn't be managed with **WAU**.
https://docs.citrix.com/en-us/citrix-workspace-app-for-windows/install.html#command-line-based-installation
You must use the installation command `AutoUpdateStream=LTSR` to update to that version!

As an example, here's my installation for my company (traditional exe-installation from SCCM):
`"%~dp0CitrixWorkspaceFullInstaller.exe" /forceinstall /silent /noreboot /includeSSON ROOTDRIVE=C:\ AutoUpdateCheck=auto DeferUpdateCount=3 EnableCEIP=False DisableSetting=3 ENABLEPRELAUNCH=True STORE0="HIDDEN;https://HIDDEN.cloud.com/Citrix/Store/discovery;On;HIDDEN" ADDLOCAL=ReceiverInside,ICA_Client,BCR_Client,USB,DesktopViewer,AM,SSON,SelfService,WebHelper`

As it is, blacklist **Citrix.Workspace*** in **WAU** and let Citrix service manage the updates!

**Reply by dlundahl91** [reply]
Thanks so much for putting time and energy into my situation.

As it is right now, we are an organization of 800 devices, and we are using WDAC (Windows Defender Application Control) and our IT Security department doesn't like when apps update by themselves, we must use Intune to do everything - which is why I was hoping we could get WAU to do this for us, but only the LTSR version.

However, you have helped me so much and taught me so much as well, I am so incredibly appreciative. You are the absolute best. 🫡

**Reply by KnifMelti** [reply]
Just a notice.
Updating from WAU is the same as letting Citrix update itself.
It's a download in SYSTEM context and executing of an installer (every application has its own) locally on every computer, surely your Security department can whitelist the Citrix engine/service, as they must have done with WAU (or they haven't)...
...with your persuading manners it's an easy task to socially be engineering that? 😉

**Reply by KnifMelti** [reply]
https://github.com/Romanitho/Winget-AutoUpdate/pull/746
